resources/migrations/000_migrations.yaml

```
@@ -12993,6 +12993,54 @@ databaseChangeLog:
             tableName: metabase_database
             columnName: details
 
+  - changeSet:
+      id: v45.00-044
+      author: metamben
+      comment: Added 0.45.0 -- create app permission graph revision table
+      changes:
+        - createTable:
+            tableName: app_permission_graph_revision
+            remarks: 'Used to keep track of changes made to app permissions.'
+            columns:
+              - column:
+                  name: id
+                  type: int
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: before
+                  type: ${text.type}
+                  remarks: 'Serialized JSON of the apps graph before the changes.'
+                  constraints:
+                    nullable: false
+              - column:
+                  name: after
+                  type: ${text.type}
+                  remarks: 'Serialized JSON of the apps graph after the changes.'
+                  constraints:
+                    nullable: false
+              - column:
+                  name: user_id
+                  type: int
+                  remarks: 'The ID of the admin who made this set of changes.'
+                  constraints:
+                    nullable: false
+                    referencedTableName: core_user
+                    referencedColumnNames: id
+                    foreignKeyName: fk_app_permission_graph_revision_user_id
+              - column:
+                  name: created_at
+                  type: datetime
```

```suggestion
                  type: ${timestamp_type}
```

`datetime` for MySQL with default precision only has second precision which makes it more difficult to use than it needs to be -- it's pretty easy to create a bunch of objects in the same second in tests and if we're doing something like fetching the 10 most recent (with `ORDER BY created_at DESC`) we might not actually get the objects we thought we'd get